### PR TITLE
Validate visitor updates

### LINKED ISSUE #16488: Re-introduce change reverted from 15973
Re-introduce change reverted from https://github.com/OpenLiberty/open-liberty/pull/15973

### PR DESCRIPTION
**Pull request by pnicolucci**
fixes #16488

### CODE REVIEW

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

File: dev/com.ibm.ws.org.apache.jasper.el.2.2/src/org/apache/el/lang/ELSupport.java [hunk 1]
```diff
@@ -5,88 +5,91 @@
...
+                                                                                                                                 @Override
+                                                                                                                                 public Object run() {
+                                                                                                                                     return System.getProperty("com.ibm.ws.jsp.ENHANCE_EL_SUPPORT",
+                                                                                                                                                               "false");
+                                                                                                                                 }
+                                                                                                                             })).booleanValue();
+
+    public final static void throwUnhandled(Object base, Object property) throws ELException {
```
Comment (volosied) [inline dev/com.ibm.ws.org.apache.jasper.el.2.2/src/org/apache/el/lang/ELSupport.java:53]: styling here look incorrect

File: dev/com.ibm.ws.org.apache.jasper.el.2.2/src/org/apache/el/lang/ELSupport.java [hunk 2]
```diff
@@ -0,0 +1,730 @@
...
+                                                                                                                             new java.security.PrivilegedAction<Object>() {
+                                                                                                                                 @Override
+                                                                                                                                 public Object run() {
+                                                                                                                                     return System.getProperty("com.ibm.ws.jsp.ENHANCE_EL_SUPPORT",
+                                                                                                                                                               "false");
+                                                                                                                                 }
+                                                                                                                             })).booleanValue();
```
Comment (volosied) [inline dev/com.ibm.ws.org.apache.jasper.el.2.2/src/org/apache/el/lang/ELSupport.java:51]: styling here looks incorrect to me? Too far to the right?

File: dev/com.ibm.ws.org.apache.jasper.el.2.2/src/org/apache/el/lang/ELSupport.java [hunk 3]
```diff
@@ -214,41 +216,102 @@ public final static boolean equals(final Object obj0, final Object obj1)
...
      * Null and empty string are false.
+     *
+     * @param ctx       the context in which this conversion is taking place
+     * @param obj       the object to convert
+     * @param primitive is the target a primitive in which case coercion to null
+     *                      is not permitted
+     * @return the Boolean value of the object
+     * @throws ELException if object is not Boolean or String
+     */
+    public static final Boolean coerceToBoolean(final ELContext ctx, final Object obj, boolean primitive) {
```
Comment (volosied) [inline dev/com.ibm.ws.org.apache.jasper.el.2.2/src/org/apache/el/lang/ELSupport.java:302]: This should be the fixed method and it looks good.

File: dev/com.ibm.ws.org.apache.jasper.el.2.2/src/org/apache/el/lang/ELSupport.java [hunk 4]
```diff
@@ -214,41 +216,102 @@ public final static boolean equals(final Object obj0, final Object obj1)
...
+     */
+    @SuppressWarnings("unchecked")
+    public static final Enum<?> coerceToEnum(final ELContext ctx, final Object obj,
+                                             @SuppressWarnings("rawtypes") Class type) {
+        return coerceToEnum(obj, type);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static final Enum<?> coerceToEnum(final Object obj,
+                                             @SuppressWarnings("rawtypes") Class type) {
```
Comment (volosied) [inline dev/com.ibm.ws.org.apache.jasper.el.2.2/src/org/apache/el/lang/ELSupport.java:254]: These method definitions were copied from the newer ELSupport class.

**Comment by pnicolucci** [comment]
#run-libby-bot

**Review by volosied — APPROVED** [review]
Everything still looks good

**Comment by pnicolucci** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp_fat,com.ibm.ws.jsp_fat_jsp.2.3.toleration,com.ibm.ws.jsp_fat_lWAS,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.el.3.0_fat

**Comment by pnicolucci** [comment]
#run-libby-bot

**Pull request merged**